### ISSUE TITLE
Drop PHP 7.4 support

### ISSUE THREAD

**Issue by linawolf** [TYPO3-Documentation/tea#527]
This enables us to use the new language features like [Constructor Promotion](https://www.php.net/manual/en/language.oop5.decon.php#language.oop5.decon.constructor.promotion)

Public support for PHP 7.4 ends by the end of the year. Both TYPO3 v11 and v12 are able to run on PHP 8.0 and 8.1, v12 requires it.

I would suggest this change for version 3.0.0

**Comment by brotkrueml** [comment]
In principle, I am with you. As a best practise, an extension should not support EOL PHP versions.

**But:** The world outside is more complicated. By now, for example, current Debian 11 is shipped with PHP 7.4 by default. At least, [serious security bugs in PHP 7.4 should be fixed](https://wiki.debian.org/PHP#Notes_on_PHP_and_security) in this distribution also after EOL. So it may be "safe" to run PHP 7.4 after official EOL on Debian or also (paid) RedHat. Of course, one can use deb.sury.org to get the newest PHP version.

So, what we have to decide:
- Should we promote only official supported PHP versions as best practise (**opinionated**).
- Should we stick to the minimum TYPO3 requirement for PHP (**conservative**).

I am fine with both, but we should come to a consent and have a reason, why we are recommending a PHP constraint.

PS: One correction: v12 requires at least PHP 8.1, not 8.0.

**Comment by brotkrueml** [comment]
**Personally**, I tend to use a conservative constraint (support EOL PHP versions) which are also supported by the TYPO3 version. It should be in the responsibility of an extension developer then to decide when this constraint makes sense for a particular extension. Although, I mostly use current PHP versions and dropped support for older versions in my public extensions. But I also have in view the used PHP versions on Packagist for my extensions. So, it depends on the (composer) users.

As a responsible extension developer, supporting only newer PHP versions means also, that I have to support more than one (major) extension version for serious bugs and security issues. In that case, I also provide for older extension versions an update with a priority/security fix. Depending on the evolving of the code base in the newest major version, this responsibility is another burden.

So, there is no black or white, but a grey - and the developer should weigh the pros and cons. So, **my recommendation** is: support the least PHP version in EXT:tea depending on the TYPO3 version. That is for v11 and v12 PHP 7.4.

And: Using constructor property promotion in documentation examples is not a reason for **me** using PHP 8.0 as minimum constraint.

**Comment by oliverklee** [comment]
I'd like to propose we discuss at our next call how to go about PHP version support in general so we don't need to decide on a case-by-case basis.

**Comment by brotkrueml** [comment]
Yes. As I am on vacation on the next call I stated my opinion already as comment. I also propose to add a section to the documentation with the pros and cons of PHP constraints as outlined here (and on the call) as it may also be of interest by an extension author. Can do this mid-September as volunteer ;-)

**Comment by sbuerk** [comment]
I would formulate it as best practice to support all php version that supported core version supports. So if TYPO3 v11 is supported, PHP7.4 *must* be supported too.

In internal projects/extensions you are free, but public extensions should respect the core range.
